pkg/kubemark/hollow_kubelet.go

```diff
@@ -52,38 +57,96 @@ func NewHollowKubelet(
 	manifestFilePath := utils.MakeTempDirOrDie("manifest", testRootDir)
 	glog.Infof("Using %s as root dir for hollow-kubelet", testRootDir)
 
-	kubeCfg, kubeDeps := kubeletapp.SimpleKubelet(
-		client,
-		dockerClient,
-		nodeName,
-		testRootDir,
-		"",        /* manifest-url */
-		"0.0.0.0", /* bind address */
-		uint(kubeletPort),
-		uint(kubeletReadOnlyPort),
-		api.NamespaceDefault,
-		empty_dir.ProbeVolumePlugins(),
-		nil, /* tls-options */
-		cadvisorInterface,
-		manifestFilePath,
-		nil, /* cloud-provider */
-		&containertest.FakeOS{}, /* os-interface */
-		20*time.Second,          /* FileCheckFrequency */
-		20*time.Second,          /* HTTPCheckFrequency */
-		1*time.Minute,           /* MinimumGCAge */
-		10*time.Second,          /* NodeStatusUpdateFrequency */
-		10*time.Second,          /* SyncFrequency */
-		5*time.Minute,           /* OutOfDiskTransitionFrequency */
-		5*time.Minute,           /* EvictionPressureTransitionPeriod */
-		maxPods,
-		podsPerCore,
-		containerManager,
-		nil,
-	)
+	// -----------------
+	// Static config
+	// -----------------
+
+	// Do the external -> internal conversion to make sure that defaults
+	// are set for fields not overridden in NewHollowKubelet.
+	tmp := &v1alpha1.KubeletConfiguration{}
+	c := &componentconfig.KubeletConfiguration{}
+	api.Scheme.Convert(tmp, c)
+
+	// Stuff from hollow_kubelet TODO(mtaufen) Remove comment
+	c.HostnameOverride = nodeName
```

Comment:
It'd be better to have a function that does this. Something like `GetHollowKubeletConfig` or sth.